CONTRIBUTING.md

```diff
@@ -133,13 +132,31 @@ In order to keep your fork up-to-date, you need to point it to the main SuperCol
-- rebase, after which you'll need to force push: `git checkout topic/branch-description` followed by `git push origin -f`
-	- This will change the commit history and is not recommended if there is more than one person working on the branch; however, it will create a clean commit history with only your commits in the pull request.
-- don't rebase, but instead `git checkout topic/branch-description`, then `git pull`, followed by `git push origin`
-	- The result will be that your pull request will include your commits in addition to all other commits to `develop` since your branch was created (without changing the commit history).
+It is almost never a good idea to resolve merge conflicts via the GitHub interface or by merging the main branch in locally. This creates noise in commit history and makes it more difficult to perform other operations on branches later. In the SuperCollider project, the preferred way to resolve merge conflicts in a pull request is by rebasing.
+
+Rebasing (`git help rebase`) is a very useful command for four scenarios in particular:
+
+1. If you'd like to incorporate new commits on the base branch that contain relevant fixes or
+   features into your topic branch.
+2. If you'd like to resolve merge conflicts.
+3. If you'd like to change the branch your topic branch is based on. This may happen if, for instance, a maintainer requests that you make your bug fix merge request against a release branch (e.g. 3.9, 3.10) instead of develop. You will then need to rebase your topic branch onto the appropriate release branch.
+4. If you'd like to rewrite your commit history by combining or reordering some commits (not recommended for those newer to git).
+
+Rebase has an interactive mode (`git rebase -i`) which will show exactly which commits will be applied to the new base, and the order in which they will be applied. This can be very helpful when you're not completely sure what the result of a rebase will be.
+
+For scenarios (1) and (2), suppose that the current branch is `topic`, which is based on an old commit from the `develop` branch. To rebase, you can execute `git rebase -i develop`. You can examine the change list to make sure it's correct before continuing. Git will stop if it encounters a merge conflict, and give instructions on how to resolve it and continue the rebase.
+
+For scenario (3), suppose that there are three branches: `develop`; the release branch `3.10`, and the topic branch `topic` which is based on `develop`. Suppose that `topic` is currently checked out.  You would like it to be based on `3.10` instead of develop. An easy way to do this is with `git rebase -i --onto 3.10 develop topic`. Beware that you may need to resolve merge conflicts during this rebase.
```

Comment:
There's a semi-colon after `develop` that should be a comma. Also, there's one `develop` that isn't written as inline code:

> You would like it to be based on `3.10` instead of develop